evalml/pipelines/time_series_pipeline_base.py

```diff
@@ -149,19 +149,25 @@ def _add_training_data_to_X_Y(self, X, y, X_train, y_train):
     def _drop_time_index(self, X, y):
         """Helper method to drop the time index column from the data if DateTime Featurizer is not present."""
         if self.should_drop_time_index and self.time_index in X.columns:
-            X_schema = X.ww.schema
-            y_schema = y.ww.schema
             index_name = X.index.name
-            X = X.set_index(X[self.time_index])
-            X.index.name = index_name
-            y = y.set_axis(X[self.time_index])
-            X.ww.init(schema=X_schema)
+            time_index = pd.DatetimeIndex(
+                X[self.time_index],
+                freq=pd.infer_freq(X[self.time_index]),
+            )
+
+            y_schema = y.ww.schema
+            y = y.set_axis(time_index)
             y.ww.init(schema=y_schema)
+
             if X.ww.schema is not None:
-                X = X.ww.drop([self.time_index])
+                X.ww.set_index(self.time_index)
+                X = X.ww.drop(self.time_index)
```

Comment:
I thought setting the index dropped the column in woodwork as well? I might be remembering incorrectly though!

Reply:
Looking at the `_set_underlying_index()` function [here](https://woodwork.alteryx.com/en/stable/_modules/woodwork/table_accessor.html#WoodworkTableAccessor) it looks like it doesn't drop it automatically